ocrd_utils/ocrd_utils/__init__.py

```diff
@@ -784,28 +785,38 @@ def xywh_from_points(points):
     """
     return xywh_from_bbox(*bbox_from_points(points))
 
-def parse_json_string_or_file(value='{}'):    # pylint: disable=unused-argument
+def parse_json_string_with_comments(val):
+    """
+    Parse a string of JSON interspersed with #-prefixed full-line comments
+    """
+    jsonstr = re.sub('^\s*#.*$', '', val, flags=re.MULTILINE)
+    return json.loads(jsonstr)
+
+def parse_json_string_or_file(*values):    # pylint: disable=unused-argument
     """
     Parse a string as either the path to a JSON object or a literal JSON object.
 
     Empty strings are equivalent to '{}'
     """
-    ret = None
-    err = None
-    if re.fullmatch(r"\s*", value):
-        return {}
-    try:
+    ret = {}
+    for value in values:
+        err = None
+        value_parsed = None
+        if re.fullmatch(r"\s*", value):
+            continue
         try:
-            with open(value, 'r') as f:
-                ret = json.load(f)
-        except FileNotFoundError:
-            ret = json.loads(value.strip())
-        except OSError:
-            ret = json.loads(value.strip())
-        if not isinstance(ret, dict):
-            err = ValueError("Not a valid JSON object: '%s' (parsed as '%s')" % (value, ret))
-    except json.decoder.JSONDecodeError as e:
-        err = ValueError("Error parsing '%s': %s" % (value, e))
-    if err:
-        raise err       # pylint: disable=raising-bad-type
+            try:
+                with open(value, 'r') as f:
+                    value_parsed = parse_json_string_with_comments(f.read())
+            except FileNotFoundError:
+                value_parsed = parse_json_string_with_comments(value.strip())
+            except OSError:
+                value_parsed = parse_json_string_with_comments(value.strip())
+            if not isinstance(value_parsed, dict):
+                err = ValueError("Not a valid JSON object: '%s' (parsed as '%s')" % (value, value_parsed))
+        except json.decoder.JSONDecodeError as e:
+            err = ValueError("Error parsing '%s': %s" % (value, e))
+        if err:
+            raise err       # pylint: disable=raising-bad-type
+        ret = {**ret, **value_parsed}
```

Comment:
wow, haven't ever seen that before. Python _can_ be quite elegant!

Reply:
Destructuring is a ncie feature, I wish Python made even more use of it (probably does in newer versions). In JS you can write crazy concise code, define local variables based on object keys and much more.